packages/liquidator/src/liquidationStrategy.js

```diff
@@ -0,0 +1,231 @@
+const assert = require("assert");
+/**
+ * This module is responsible for sizing liquidation positions.
+ * If a defensive strategy is enabled, it will check withdraw positions
+ * to see if they require an extension rather than full liquidation.
+ * @constructor
+ * @param {object} config - State configuration, env vars, contract vars
+ * @property {number} config.whaleDefenseFundWei - Optional value which enables strategy if above 0. Specify the reserves to hold in tokens to allow for defense strategy. 0 disables.
+ * @property {number} config.defenseActivationPercent - The % past liveness a position needs to be before its extended. Optional unless whaleDefenseFundWei is enabled.
+ * @property {number} config.liquidationDeadline - Aborts the liquidation if the transaction is mined this amount of time after this time has passed
+ * @property {number} config.liquidationLiveness - Optional unless whaleDefenseFundWei is enabled. The positions liveness duration, set by emp contract.
+ * @property {number} config.minSponsorSize - Emp minimum sponsor position size in tokens.
+ * @param {object} deps - Library dependencies
+ * @property {object} deps.toBN - toBN function
+ * @property {object} deps.BN - BN utilities
+ * @callback emit - A way to emit logs to an external logger
+ * @param {string} severity - Log level severity
+ * @param {object} data - Any data you want to be sent in log
```

Comment:
Tried to write this in JSDoc style but not totally fluent in this. Not sure if theres some tooling around this that is useful

Reply:
Looks good! No tooling around it that I know...We normally just try to have a consistent way to structuring it.

One small nit is that your line lengths are longer than the 120 limit we have defined in the styling guide. I'd say that the initial comment between lines 3 and 5 could be longer (up to 120 chars) and then the subsequent comments should be shorter.

Also we have not uses the property structure before; I think it makes sense though. For example here https://github.com/UMAprotocol/protocol/blob/3129fc83480531d31ba64af5bb10ac061f93e6bf/packages/monitors/src/CRMonitor.js#L16 we give an "example" object. Curious on which pattern we prefer.